### PR TITLE
Add UTs for C#11 "new lines in interpolated strings" feature (non SonarWay rules)

### PR DESCRIPTION
**Pull request by mary-georgiou-sonarsource**

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/NormalizeStringsToUppercaseTest.cs [hunk 1]
```diff
@@ -29,5 +29,16 @@ public class NormalizeStringsToUppercaseTest
         public void NormalizeStringsToUppercase() =>
             new VerifierBuilder<NormalizeStringsToUppercase>().AddPaths("NormalizeStringsToUppercase.cs")
                 .Verify();
+
+#if NET
+
+        [TestMethod]
+        public void NormalizeStringsToUppercase_CSharp11() =>
+            new VerifierBuilder<NormalizeStringsToUppercase>().AddPaths("NormalizeStringsToUppercase.CSharp11.cs")
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/NormalizeStringsToUppercaseTest.cs:37]: The guide line for the VerifierBuilder is to create a field
```cs
private readonly VerifierBuilder builder = new VerifierBuilder<DatabasePasswordsShouldBeSecure>();
```
and use it in the tests.
```suggestion
            builder.AddPaths("NormalizeStringsToUppercase.CSharp11.cs")
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs [hunk 2]
```diff
@@ -22,3 +22,21 @@ public class FooLessThanFiveCharacters
 
     public static readonly string NameReadonly = """foo""";
 }
+
+public class FooNonCompliantStringInterpolation
+{
+    static string GetName() => "Bar";
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs:28]: Needs at least 5 characters:
```suggestion
    static string GetName() => "BarBar";
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs [hunk 3]
```diff
@@ -22,3 +22,21 @@ public class FooLessThanFiveCharacters
...
+
+    public string NameOne = $"{
+        GetName()
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs:31]:
```suggestion
        "BarBar"
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs [hunk 4]
```diff
@@ -22,3 +22,21 @@ public class FooLessThanFiveCharacters
...
+        }"; // FN - {{Define a constant instead of using this literal '""Bar""' 4 times.}}
+
+    public string NameTwo = $"{
+        GetName()
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs:35]:
```suggestion
        "BarBar"
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs [hunk 5]
```diff
@@ -22,3 +22,21 @@ public class FooLessThanFiveCharacters
...
+        }"; // FN
+
+    public static string NameThree = "Bar"; // FN
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs:38]:
```suggestion
    public static string NameThree = "BarBar"; // FN
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs [hunk 6]
```diff
@@ -22,3 +22,21 @@ public class FooLessThanFiveCharacters
...
+
+    public static readonly string NameReadonly = $"{GetName()}"; // FN
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs:40]:
```suggestion
    public static readonly string NameReadonly = $"{"BarBar"}"; // FN
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringOperationWithoutCulture.CSharp11.cs [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    public class StringOperationWithoutCulture
+    {
+        void TestInterpolatedStrings()
+        {
+            int foo = 10;
+            string s = $"Hello {foo switch
+                {
+                    > 0 => "FOO",
+                    _ => "Bar",
+                }} ".ToUpper(); // Noncompliant {{Define the locale to be used in this string operation.}}
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringOperationWithoutCulture.CSharp11.cs:18]:
```suggestion
                }}".ToUpper(); // Noncompliant {{Define the locale to be used in this string operation.}}
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringOperationWithoutCulture.CSharp11.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Globalization;
+using System.Linq.Expressions;
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringOperationWithoutCulture.CSharp11.cs:5]:
```suggestion
```

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NormalizeStringsToUppercase.CSharp11.cs [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+                < 0 => "Negative",
+                _ => "Zero",
+            }}".ToLower(); // Compliant
+
+            var s2 = $"The score: {score} is {score switch
+            {
+                > 0 => "Positive",
+                < 0 => "Negative",
+                _ => "Zero",
+            }}".ToLowerInvariant(); // Noncompliant
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NormalizeStringsToUppercase.CSharp11.cs:22]: Noncompliant? According to StringOperationWithoutCulture.cs test cases, it is the other way around: `ToLowerInvariant` is compliant and `ToLower` is NonCompliant.

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
In one test case "Noncompliant" and "Compliant" are mixed up.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs (hunk 3 above)
Comment (mary-georgiou-sonarsource) [reply analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs]: Thanks a lot for catching this. 🤦

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/StringOperationWithoutCultureTest.cs [hunk 10]
```diff
@@ -28,5 +28,16 @@ public class StringOperationWithoutCultureTest
         [TestMethod]
         public void StringOperationWithoutCulture() =>
             new VerifierBuilder<StringOperationWithoutCulture>().AddPaths("StringOperationWithoutCulture.cs").Verify();
+
+#if NET
+
+        [TestMethod]
+        public void StringOperationWithoutCulture_CSharp11() =>
+            new VerifierBuilder<StringOperationWithoutCulture>().AddPaths("StringOperationWithoutCulture.CSharp11.cs")
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/StringOperationWithoutCultureTest.cs:36]: Use `builder ` field

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs [hunk 11]
```diff
@@ -22,3 +22,17 @@ public class FooLessThanFiveCharacters
...
     public static readonly string NameReadonly = """foo""";
 }
+
+public class FooNonCompliantStringInterpolation
+{
+    public string NameOne = $"{"BarBar" // Noncompliant {{Define a constant instead of using this literal 'BarBar' 4 times.}}
+        }";
+
+    public string NameTwo = $"{"BarBar" // Secondary
+        }";
```
Comment (martin-strecker-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/StringLiteralShouldNotBeDuplicated.CSharp11.cs:32]: Just to change the formatting a bit between the four assignments:

```suggestion
    public string NameTwo = $"{
            "BarBar" // Secondary
        }";
```

**Review by martin-strecker-sonarsource — CHANGES_REQUESTED** [review]
One more VerfierBuilder and a small change in one test.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**